File: Cours-POO/Controle-20240206_Gestion-Pharmacie/Implementation-Cpp/Medicament.cpp
#include "Medicament.h"

Medicament::Medicament(string nomMed, double prix, int quantite) {
    this->nomMed = nomMed;
    this->prix = prix;
    this->quantite = quantite;
}

string Medicament::getNomMed() const {
    return this->nomMed;
}

void Medicament::setNomMed(string nomMed) {
    this->nomMed = nomMed;
}

double Medicament::getPrix() const {
    return this->prix;
}

void Medicament::setPrix(double prix) {
    this->prix = prix;
}

int Medicament::getQuantite() const {
    return this->quantite;
}

void Medicament::setQuantite(int quantite) {
    this->quantite = quantite;
}

void Medicament::affichageMed() const {
    cout << "Medicament : " << this->nomMed << " - Prix : " << this->prix << " - Quantité : " << this->quantite << endl;
}

File: Cours-POO/Controle-20240206_Gestion-Pharmacie/Implementation-Java/Medicament.java
public class Medicament {
    private String nomMed;
    private double prix;
    private int quantite;

    public Medicament(String nomMed, double prix, int quantite) {
        this.nomMed = nomMed;
        this.prix = prix;
        this.quantite = quantite;
    }

    public String getNomMed() {
        return this.nomMed;
    }

    public void setNomMed(String nomMed) {
        this.nomMed = nomMed;
    }

    public double getPrix() {
        return this.prix;
    }

    public void setPrix(double prix) {
        this.prix = prix;
    }

    public int getQuantite() {
        return this.quantite;
    }

    public void setQuantite(int quantite) {
        this.quantite = quantite;
    }

    public void affichageMed() {
        System.out.println("Medicament : " + this.nomMed + " - Prix : " + this.prix + " - Quantité : " + this.quantite);
    }
}